src/runtime-rs/crates/runtimes/common/src/types/mod.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+mod trans_from_shim;
+mod trans_into_shim;
+
+use std::fmt;
+
+use kata_types::mount::Mount;
+
+#[derive(Debug, Clone)]
+pub enum Request {
+    CreateContainer(ContainerConfig),
+    CloseIo(ProcessID),
+    DeleteProcess(ProcessID),
+    Exec(ExecProcessRequest),
+    KillProcess(KillRequest),
+    Wait(ProcessID),
+    StartProcess(ProcessID),
+    StateProcess(ProcessID),
+    Shutdown(ShutdownRequest),
+    Pause(String),
+    Resume(String),
+    ResizePty(ResizePtyRequest),
+    Stats(String),
+    Update(UpdateRequest),
+    Pid,
+    Connect(String),
+}
+
+impl fmt::Display for Request {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Response {
+    CreateContainer(ShimPid),
+    CloseIo,
+    DeleteProcess(ProcessStateInfo),
+    Exec,
+    KillProcess,
+    Wait(ProcessExitStatus),
+    StartProcess(ShimPid),
+    StateProcess(ProcessStateInfo),
+    Shutdown,
+    Pause,
+    Resume,
+    ResizePty,
+    StatsResponse(StatsInfo),
+    Update,
+    Pid(ShimPid),
+    Connect(ShimPid),
+}
+
+impl std::fmt::Display for Response {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ProcessType {
+    Container,
+    Exec,
+}
+
+#[derive(Clone, Debug)]
+pub struct ProcessID {
+    pub cid: String,
+    pub eid: String,
+    pub process_type: ProcessType,
+}
+
+impl fmt::Display for ProcessID {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+impl ProcessID {
+    pub fn new(cid: &str, eid: &str) -> Self {
```

Comment:
- Could you add some unit tests for this function (and for any other new ones)?
- This function doesn't perform any validation on the params so is that up to the caller, or a callee?

Reply:
Thanks for your review, add validate for cid and eid.